src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -323,6 +327,38 @@ def lookup_job_info(self, job_ids: List[str]) -> dict:
             infos[error_id] = "does_not_exist"
         return infos
 
+    def lookup_jobs_by_cell_id(self, cell_id_list=None):
+        """
+        Fetch job states for jobs with a cell_id in cell_id_list
+        Batch job cell IDs are calculated using the `batch_cell_ids` method
+        Returns a dictionary of job states keyed by job ID and a mapping of
+        cell IDs to the list of job IDs associated with the cell.
+        """
+        if not cell_id_list:
+            raise ValueError(CELLS_NOT_PROVIDED_ERR)
+
+        jobs_to_lookup = list()
+        cell_to_job_mapping = {id: set() for id in cell_id_list}
```

Comment:
Should this just be maintained by the jobmanager? When a new job is created, add it to `_running_jobs` and index it in the `cell_to_job_mapping`? Might make it easier than rebuilding it each time.

Unless it's not going to be needed very often.

Reply:
that's a good idea. I will implement that.